client/App.jsx

```diff
@@ -80,6 +81,23 @@ class App extends Component {
     this.setState({ drawerShowing: null });
   }
 
+  beginQrScan(imageBufferData) {
+    return new Promise((resolve, reject) => {
+      let qrWorker = new Worker('/qr-worker.js');
+      qrWorker.postMessage(imageBufferData);
+      qrWorker.onmessage = (qrData) => {
+        if (qrData.data.result) {
+          resolve(qrData.data.result);
+        } else {
+          reject(qrData.data.error);
+        }
+      };
+    }).then((qrData) => {
+      console.log(qrData);
+      return qrData;
+    });
```

Comment:
To remove later. Some fairly verbose logging should be present in the "starting point" for the QR/web worker exercise, so the cross-thread communication is 100% clear.